src/test/java/uk/gov/homeoffice/digital/sas/timecard/validators/timeentry/TimeEntryValidatorTest.java

```diff
@@ -236,6 +325,53 @@ void validate_timeEntryIdsAreTheSameAndTimesClash_noErrorReturned() {
 
     // endregion
 
+    // region dynamic_payload
+
+    // existing: 08:00-, new: 08:00-
+    @Test
+    void validate_newStartTimeIsTheSameAsExistingStartTimeWithNoEndTimes_dynamicPayloadSet() {
+        var time = LocalDateTime.of(
+            2022, 1, 1, 8, 0, 0);
+
+        saveEntryAndFlushDatabase(createTimeEntry(
+            OWNER_ID_1,
+            getAsDate(time)));
+
+        var newStartTime = getAsDate(time);
+
+        var timeEntryNew = createTimeEntry(OWNER_ID_1, newStartTime);
+
+        Throwable thrown = catchThrowable(() -> saveEntryAndFlushDatabase(timeEntryNew));
+        var constraintViolationException = (ConstraintViolationException) thrown;
+        var hibernateConstraintViolation = constraintViolationException.getConstraintViolations().iterator().next().unwrap(
+            HibernateConstraintViolation.class);
+        var dynamicPayload = hibernateConstraintViolation.getDynamicPayload(ArrayList.class);
+
+        var payload = ((ArrayList<JSONObject>) dynamicPayload).get(0);
```

Comment:
we should assert the list is not empty before getting the first index

Reply:
Done 👍